dask_match/core.py

```diff
@@ -409,25 +409,45 @@ class Projection(Elemwise):
     operation = operator.getitem
 
     def _divisions(self):
-        return self.frame.divisions
+        return self.operand("frame").divisions
 
     @property
     def _meta(self):
-        return self.frame._meta[self.columns]
+        return self.operand("frame")._meta[self.columns]
 
     def _layer(self):
         return {
-            (self._name, i): (operator.getitem, (self.frame._name, i), self.columns)
+            (self._name, i): (operator.getitem, (self.operand("frame")._name, i), self.columns)
             for i in range(self.npartitions)
         }
 
     def __str__(self):
-        base = str(self.frame)
+        base = str(self.operand("frame"))
         if " " in base:
             base = "(" + base + ")"
         return f"{base}[{repr(self.columns)}]"
 
 
+class Index(Elemwise):
+    """Column Selection"""
+
+    _parameters = ["frame"]
+    operation = getattr
+
+    def _divisions(self):
+        return self.operand("frame").divisions
+
+    @property
+    def _meta(self):
+        return self.operand("frame")._meta.index
+
+    def _layer(self):
+        return {
+            (self._name, i): (getattr, (self.operand("frame")._name, i), "index")
```

Comment:
Why was `.operand` necessary here?  This seems unpleasant to do for all operands.  I'm optimizing pretty hard here for "pleasant to work with"

Reply:
>Why was .operand necessary here? This seems unpleasant to do for all operands. I'm optimizing pretty hard here for "pleasant to work with"

I'm totally on board with the "pleasant to work with" goal, and so the `operand` proposal may not be the best solution. Overall, I found the current practice of accessing parameters/operands as attributes (e.g. `self.<operand>`) to be surprisingly problematic, and eventually rationalized the idea to forbid the practice altogether.  I realize that `self.<operand>` pattern *can* be supported just fine, but I'm not quite convinced that it is worth the potential for pain (though my mind is open).